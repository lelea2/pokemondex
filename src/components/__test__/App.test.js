import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import ConnectedApp, { App } from '../App'

describe('Component: App', () => {
  it('App - render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App fetchPokemonList={() => null} pokemonList={{ data: [], loading: false }} />, div);
  });

  it('App - snapshot', () => {
    const test = renderer.create(
      <App fetchPokemonList={() => null}
        pokemonList={{ data: [{ name: 'Pikachu', url: '' }, { name: 'Pikacha', url: '' }], loading: false }} />
    ).toJSON();
    expect(test).toMatchSnapshot();
  });
});