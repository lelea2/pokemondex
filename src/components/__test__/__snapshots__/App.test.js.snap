// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App App - snapshot 1`] = `
<div
  className="Container"
>
  <div
    className="title"
  >
    <h1>
      PokemonDex
    </h1>
    <form
      className="searchbox"
      onSubmit={[Function]}
    >
      <input
        className="inputSearch"
        onChange={[Function]}
        placeholder="search"
        value=""
      />
      <a
        onClick={[Function]}
      >
        <i>
          Reset Text
        </i>
      </a>
    </form>
  </div>
  <div
    className="App"
  >
    <div>
      <div
        className="reset-region"
      >
        <button
          className="reset"
          onClick={[Function]}
        >
          Reset Pokemon Detail View
        </button>
      </div>
      <section
        className="poke-list"
      >
        <button
          className="poke-cell"
          onClick={[Function]}
        >
          <img
            alt="Pikachu"
            height="50"
            src="https://img.pokemondb.net/artwork/Pikachu.jpg"
            width="50"
          />
        </button>
        <button
          className="poke-cell"
          onClick={[Function]}
        >
          <img
            alt="Pikacha"
            height="50"
            src="https://img.pokemondb.net/artwork/Pikacha.jpg"
            width="50"
          />
        </button>
      </section>
      <section
        className="detail-view"
      >
        <img
          alt=""
          className="sprite-image"
          src={null}
        />
        <div
          className="data-wrapper"
        />
      </section>
    </div>
  </div>
</div>
`;
